pkgs/test_core/lib/src/executable.dart

```diff
@@ -48,7 +48,11 @@ void completeShutdown() {
     signalSubscription = null;
   }
   isShutdown = true;
-  stdinLines.cancel(immediate: true);
+
+  // Applications can't access stdin on Fuchsia.
+  if (!Platform.isFuchsia) {
```

Comment:
After some further thought, what do you think about handling this case at the definitio of `stdInLines`.

https://github.com/dart-lang/test/blob/8716882291a62aeb0058aaf544549435439aed7e/pkgs/test_core/lib/src/util/io.dart#L68

We could do something like

```dart
final stdinLines = StreamQueue(
    Platform.isFuchsia ? Stream<String>.empty() : lineSplitter.bind(stdin));
```

As is I think we're relying on this `.cancel` call being the _only_ interaction with `stdin`, which won't help us if we hit one of the other code paths that tries to read it. I anticipate that an empty stream there could break us in other subtle ways, but probably safer anyway.

Reply:
Adding it in the definition of stdinLines is indeed a better idea. Updated. Thanks.